cmd/transfer-pvc/progress.go

```diff
@@ -0,0 +1,555 @@
+package transfer_pvc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type rsyncLogStream struct {
+	pvc        types.NamespacedName
+	podLabels  map[string]string
+	restCfg    *rest.Config
+	stdout     chan string
+	stderr     chan string
+	err        chan error
+	progress   *Progress
+	outputFile *string
+}
+
+func NewRsyncLogStream(restCfg *rest.Config, pvc types.NamespacedName, labels map[string]string, output string) LogStreams {
```

Comment:
Along with the one below, I think we should find a lib type place for this. Are we thinking that bakube might work?